style/servo/restyle_damage.rs

```diff
@@ -89,41 +99,35 @@ fn compute_damage(old: &ComputedValues, new: &ComputedValues) -> ServoRestyleDam
     // This should check every CSS property, as enumerated in the fields of
     // https://doc.servo.org/style/properties/struct.ComputedValues.html
 
+    // Some properties establish a stacking context when they are set to a non-initial value.
+    // In that case, the damage is only set to `ServoRestyleDamage::REPAINT` because we don't
+    // need to rebuild stacking contexts when the style changes between different non-initial
+    // values. However, here we check if the change is between the initial value and another
+    // one, in order to also force rebuilding stacking contexts.
+    // TODO: `z-index` doesn't always apply, so this restyle damage may be unnecessary.
+    let rebuild_stacking_context_extra = || {
+        old.get_position().z_index.is_auto() != new.get_position().z_index.is_auto() ||
+            (old.get_effects().opacity == 1.0) != (new.get_effects().opacity == 1.0) ||
+            (old.get_effects().mix_blend_mode == MixBlendMode::Normal) !=
+                (new.get_effects().mix_blend_mode == MixBlendMode::Normal) ||
+            (old.get_svg().clip_path == ClipPath::None) !=
+                (new.get_svg().clip_path == ClipPath::None)
+    };
```

Comment:
I think the logic here ought to "OR together" every old property, separately "OR together" every new property, and then compare whether the overall result of "establishes a stacking context" has changed. This would allow us to avoid rebuilding stacking contexts in cases where multiple properties change but cancel each other out.

Reply:
This may also need to be changed in the generated macro?

Reply:
This is a good point, but I don't think we can do it like that, because here we don't know if `z-index` is actually applying.

For example, if the style changes from `z-index: 1` to `opacity: 0.5` then both the old and the new styles have non-initial values for properties that can establish a stacking context, but only the new style actually establishes it if `z-index` doesn't apply.

Maybe the style damage should be computed in Servo where we have the real logic for knowing whether a style establishes a stacking context or not.

Reply:
I have updated the comment to clarify this

Reply:
Actually, any `z-index` change needs to rebuild the stacking context tree. So it's no longer handled here, and I have applied your suggestion for the remaining properties.